superset-frontend/src/SqlLab/components/EstimateQueryCostButton/index.tsx

```diff
@@ -26,24 +25,28 @@ import Loading from 'src/components/Loading';
 import ModalTrigger from 'src/components/ModalTrigger';
 import { EmptyWrapperType } from 'src/components/TableView/TableView';
 
-const propTypes = {
-  dbId: PropTypes.number.isRequired,
-  schema: PropTypes.string.isRequired,
-  sql: PropTypes.string.isRequired,
-  getEstimate: PropTypes.func.isRequired,
-  queryCostEstimate: PropTypes.Object,
-  selectedText: PropTypes.string,
-  tooltip: PropTypes.string,
-  disabled: PropTypes.bool,
-};
-const defaultProps = {
-  queryCostEstimate: [],
-  tooltip: '',
-  disabled: false,
-};
+interface EstimateQueryCostButtonProps {
+  dbId: number;
+  schema: string;
+  sql: string;
+  getEstimate: Function;
+  queryCostEstimate: Record<string, any>;
+  selectedText?: string;
+  tooltip?: string;
+  disabled: boolean;
+}
 
-const EstimateQueryCostButton = props => {
-  const { cost } = props.queryCostEstimate;
+const EstimateQueryCostButton = ({
+  dbId,
+  schema,
+  sql,
+  getEstimate,
+  queryCostEstimate = [],
```

Comment:
this probably should be 
```suggestion
  queryCostEstimate = {},
```